### PR TITLE
#1223 Tests for `OptimizeMojo` caching

### PR DESCRIPTION
**Pull request by volodya-lombrozo**
I've added several tests for the future `OptimiseMojo` caching:
1) We expect that `OptimizeMojo` will get `xmir` file from the cache (`getsAlreadyOptimizedResultsFromCache`)
2) We expect that `OptimizeMojo` will create a file in the cache if an optimisation result is absent in the cache (`savesOptimizedResultsToCache`)

Original issue: #1223

### CODE REVIEW

File: eo-maven-plugin/src/test/java/org/eolang/maven/OptimizeMojoTest.java [hunk 1]
```diff
@@ -129,6 +132,104 @@ void optimizesIfExpired(@TempDir final Path temp) throws Exception {
         );
     }
 
+    /**
+     * Test case for #1223.
+     *
+     * @param temp Temporary test directory.
+     * @throws Exception if unexpected error happened.
+     * @todo #1223:90min remove '@Disabled' annotation when caching of already optimized xmir files
```
Comment (mximp) [inline eo-maven-plugin/src/test/java/org/eolang/maven/OptimizeMojoTest.java:140]: @volodya-lombrozo maybe it's worth to create puzzle for caching implementation?

Comment (volodya-lombrozo) [reply eo-maven-plugin/src/test/java/org/eolang/maven/OptimizeMojoTest.java]: @mximp I've updated the puzzle description. Please, check.

**Review by mximp — APPROVED** [review]

**Comment by volodya-lombrozo** [comment]
@yegor256 merge, please.

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**